src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -3919,11 +3919,11 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
 
                 // Handle op3
-                if (op3->IsVectorZero() && op1->IsMaskAllBitsSet())
+                if (op3->IsVectorZero() && op1->IsMaskAllBitsSet() && op2->IsEmbMaskOp())
                 {
                     // When we are merging with zero, we can specialize
                     // and avoid instantiating the vector constant.
-                    // Do this only if op1 was AllTrueMask
+                    // Do this only if op1 was AllTrueMask and op2 is embedded.
                     MakeSrcContained(node, op3);
                 }
```

Comment:
I don't think this is the right fix. The `zero` (and in fact any constant or other value here) is still containable.

`op1` is `AllBitsSet` so therefore nothing from `op3` can ever be selected, so it is "unused". This means it is valid to drop the `sel` entirely and just emit `op2` directly.

The only time we should be hitting this path is when `op2` is an operation that requires a mask to be specified (even if unused) or some manually written user code in `minopts`. For the latter, it's still fine to contain the zero constant and just use the op2 register for both the input and output.

Reply:
> `op1` is `AllBitsSet` so therefore nothing from `op3` can ever be selected, so it is "unused". This means it is valid to drop the `sel` entirely and just emit `op2` directly.

Updated to do this instead.

Required the embedded HWIntrinsic check, otherwise lowering goes into an infinite loop adding and removing conditional selects.